### PR TITLE
Migration to Navigation Compose

### PR DESCRIPTION
**Pull request by chrisbanes**
This PR removes (nearly) all Fragments from Tivi*, and migrates everything to Compose Navigation. There are no animations (yet), but that will come.

\* _The only fragment left is a `PreferenceFragment` which I likely won't migrate over to Compose for a while._

### CODE REVIEW

File: ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetailsViewModel.kt [hunk 1]
```diff
@@ -54,7 +55,12 @@ class EpisodeDetailsViewModel @AssistedInject constructor(
     private val removeEpisodeWatch: RemoveEpisodeWatch,
     private val logger: Logger,
     private val snackbarManager: SnackbarManager
-) : ReduxViewModel<EpisodeDetailsViewState>(initialState) {
+) : ReduxViewModel<EpisodeDetailsViewState>(
+    EpisodeDetailsViewState(
+        // The string "episodeId" is the name of the argument in the route
+        episodeId = savedStateHandle.get<String>("episodeId")!!.toLong()
```
Comment (ianhanniballake) [inline ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetailsViewModel.kt:61]: Note that [defining a type on your argument](https://developer.android.com/jetpack/compose/navigation#nav-with-args) will mean this is already a Long, avoiding the need to cast.

Comment (chrisbanes) [reply ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetailsViewModel.kt:61]: Hmmm I got a `ClassCastException` when I tried that earlier, which is why I added the explicit conversion. I'll double check tomorrow.

Comment (chrisbanes) [reply ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetailsViewModel.kt:61]: Ohhh, just read the link. Makes sense, I'll move over to that.

Comment (chrisbanes) [reply ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetailsViewModel.kt:61]: Done in #778

**Pull request merged**